### PR TITLE
Add restriction on inverse and derivative

### LINKED ISSUE #2837: Missing explicit ban on providing inconsistent derivatives or inverses
For a model to be portable between tools and even different versions of the same tool, it is necessary that tools can rely on provided derivatives and inverses being defined in a mathematically consistent manner.  Hence, there should probably be an explicit ban on providing inconsistent derivatives or inverses of functions.  Otherwise, users might think it's fine to provide things that will trick the symbolic processing of a particular version of a particular tool into computing some crazy stuff.

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #2837

### CODE REVIEW

File: chapters/functions.tex [hunk 1]
```diff
@@ -1515,10 +1538,9 @@ \section{Declaring Inverses of Functions}\label{declaring-inverses-of-functions}
 More than one inverse can be defined within the same \lstinline!inverse! annotation.
 Several inverses are separated by commas.
 
-\begin{nonnormative}
-The inverse requires that for all valid values of the input arguments of \lstinline!$f_2$($\ldots$, y, $\ldots$)! and $u_k$ being calculated as \lstinline!$u_k$ := $f_2$($\ldots$, y, $\ldots$)! implies
-the equality \lstinline!y = $f_1$($\ldots$, $u_k$, $\ldots$)! up to a certain precision.
-\end{nonnormative}
+The function in the \lstinline!inverse! annotation must be an actual inverse.
+This requires that for all valid values of the input arguments of \lstinline!$f_2$($\ldots$, y, $\ldots$)! and $u_k$ being calculated as \lstinline!$u_k$ := $f_2$($\ldots$, y, $\ldots$)! implies
```
Comment (eshmoylova) [inline chapters/functions.tex:1542]: I find this sentence very hard to parse. Maybe:
```suggestion
This requires that $f_2$ is such that, if $u_k$ is calculated as \lstinline!$u_k$ := $f_2$($\ldots$, y, $\ldots$)!, the equality \lstinline!y = $f_1$($\ldots$, $u_k$, $\ldots$)! is satisfied up to a certain precision, for all valid values of the input arguments of \lstinline!$f_2$($\ldots$, y, $\ldots$)!.
```

**Review by eshmoylova — CHANGES_REQUESTED** [review]

File: chapters/functions.tex [hunk 2]
```diff
@@ -1176,13 +1176,36 @@ \subsection{Using the Derivative Annotation}\label{using-the-derivative-annotati
...
 There may be multiple restrictions on the derivative, in which case they must all be satisfied. The restrictions also imply that some derivatives of some inputs are excluded from the call of the derivative (since they are not necessary).
 A function may supply multiple derivative functions subject to different restrictions, the first one that can be used (i.e.\ satisfying the restrictions) will be used for each call.
 
 \begin{nonnormative}
 This means that the most restrictive derivatives should be written first.
 \end{nonnormative}
+\begin{nonnormative}
+A straightforward way of explaining a valid derivative is given by this model.
+The function value, \lstinline!x1!, should up to numerical precision match the integral of the derivative, \lstinline!x2!.
+Note that in practive tools may over-optimize this.
```
[the remark this answers is not in the document]
Comment (HansOlsson) [reply chapters/functions.tex]: Thanks for catching this.

File: chapters/functions.tex [hunk 3]
```diff
@@ -1176,13 +1176,36 @@ \subsection{Using the Derivative Annotation}\label{using-the-derivative-annotati
...
 These restrictions are defined using the following optional attributes: \lstinline!order! (only a restriction if $\text{\lstinline!order!} > 1$, the default for \lstinline!order! is 1), \lstinline!noDerivative!, and \lstinline!zeroDerivative!.
-The given derivative-function can only be used to compute the derivative of a function call if these restrictions are satisfied.
+The given derivative-function must be a valid derivative if these restrictions are satisfied, and can thus be used to compute the derivative in those cases.
 There may be multiple restrictions on the derivative, in which case they must all be satisfied. The restrictions also imply that some derivatives of some inputs are excluded from the call of the derivative (since they are not necessary).
 A function may supply multiple derivative functions subject to different restrictions, the first one that can be used (i.e.\ satisfying the restrictions) will be used for each call.
 
 \begin{nonnormative}
 This means that the most restrictive derivatives should be written first.
 \end{nonnormative}
+\begin{nonnormative}
```
[the remark this answers is not in the document]
Comment (HansOlsson) [reply chapters/functions.tex:1186]: Hmm... I'm not sure, and I think nonnormative makes sense here. We should likely develop better guidelines.

To me example is for general examples, whereas the nonnormative text is more about explanations - and the main part of this is that it explains what a derivative means.

**Review by eshmoylova — APPROVED** [review]
Looks good to me.

File: chapters/functions.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (eshmoylova) [reply chapters/functions.tex:1186]: OK, I agree that nonnormative makes more sense here.

File: chapters/functions.tex [hunk 4]
```diff
@@ -1176,13 +1176,36 @@ \subsection{Using the Derivative Annotation}\label{using-the-derivative-annotati
...
+A straightforward way of explaining a valid derivative is given by this model.
+The function value, \lstinline!x1!, should up to numerical precision match the integral of the derivative, \lstinline!x2!.
+Note that in practice tools may over-optimize this.
```
Comment (henrikt-ma) [inline chapters/functions.tex:1189]: To say _over-optimize_ sounds a little like the tools would be doing it wrong.  I'd rather say this:
> Note that tools are not required to use the provided derivative, and might solve the equations completely without numeric integration.

**Review by henrikt-ma — COMMENTED** [review]
Adding a comment after the fact of merging, and will create separate PR for further discussion.
